### PR TITLE
change the memory allocator of the HNSW in stl to a custom memory allocator

### PR DESCRIPTION
**Pull request by inabao**

### CODE REVIEW

File: src/utils.h [hunk 1]
```diff
@@ -20,12 +20,27 @@
...
 
+#include "default_allocator.h"
 #include "spdlog/spdlog.h"
 #include "vsag/errors.h"
 #include "vsag/expected.hpp"
 
 namespace vsag {
 
+template <typename T>
+using unordered_set =
```
Comment (jiaweizone) [inline src/utils.h]: Use `UnorderedSet` explicitly to distinguish it from the std lib

File: src/utils.h [hunk 2]
```diff
@@ -20,12 +20,27 @@
...
+    std::unordered_set<T, std::hash<T>, std::equal_to<T>, vsag::AllocatorWrapper<T>>;
+
+template <typename T>
+using vector = std::vector<T, vsag::AllocatorWrapper<T>>;
```
Comment (jiaweizone) [inline src/utils.h]: ditto: vector -> Vector

File: src/utils.h [hunk 3]
```diff
@@ -20,12 +20,27 @@
...
+
+template <typename KeyType, typename ValType>
+using unordered_map = std::unordered_map<KeyType,
```
Comment (jiaweizone) [inline src/utils.h]: ditto

File: src/utils.h (hunk 1 above)
Comment (inabao) [reply src/utils.h]: fixed

File: src/utils.h (hunk 2 above)
Comment (inabao) [reply src/utils.h]: fixed

File: src/utils.h (hunk 3 above)
Comment (inabao) [reply src/utils.h]: fixed

**Review by wxyucs — APPROVED** [review]
lgtm

**Review by jiaweizone — APPROVED** [review]
LGTM

**Pull request merged**